core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/InvokerReactive.scala

```diff
@@ -263,7 +263,10 @@ class InvokerReactive(
                 }
 
                 val context = UserContext(msg.user)
-                val activation = generateFallbackActivation(msg, response)
+                val activation = generateFallbackActivation(
+                  msg,
+                  response,
+                  Parameters(WhiskActivation.kindAnnotation, JsString(Exec.UNKNOWN)))
```

Comment:
since it both cases of generating a fall back activation, the action is not available, i favor not changing the function sigature and adding the unknow annotation directly in `generateFallbackActivation` instead of at the two call sites.

Reply:
@rabbah although I originally suggested the approach to extend the signature of `generateFallbackActivation()` to allow for passing additional annotations in #4615, I now agree with your assessment.

`generateFallbackActivation()` is used exactly two times within the same compilation unit in close proximity to its definition. So it's fine to add the `kind = unknown` annotation directly within `generateFallbackActivation()`.

@jiangpengcheng I suggest to add a comment to `generateFallbackActivation()` why the `kind = unknown` annotation is set, i.e. in order to support user metrics because the metrics record requires a kind.